app/javascript/controllers/direct_file_uploader_controller.js

```diff
@@ -0,0 +1,93 @@
+import { Controller } from "@hotwired/stimulus"
+import { DirectUpload } from "@rails/activestorage";
+
+export default class extends Controller {
+  connect() {
+    // Attach an event listener to the file input
+    this.input = this.element.querySelector('input[type=file][data-direct-upload-url]:not([disabled])')
+  }
+
+  handleFileSelect(event) {
+    const fileInput = event.target;
+    const titleEl = document.getElementById('csv_upload_title')
+
+    // if no title is given, put a title in from the provided file
+    if (titleEl && (!titleEl.value || titleEl.value.trim() === '')) {
+      // get the filename, without the extension
+      const fileName = fileInput.value.split(/(\\|\/)/g).pop().replace('.csv', '')
+      titleEl.value = fileName
+    }
+
+    Array.from(fileInput.files).forEach(file => this.uploadFile.call(this, file, fileInput))
+  }
+
+  uploadFile(file, fileInput) {
+    const url = this.input.dataset.directUploadUrl;
```

Comment:
I just realised we don't need to set `this.input` after all:

```suggestion
    const url = fileInput.dataset.directUploadUrl;
```

Then we can also delete `connect()` entirely.